cli/cmd/push/push.go

```diff
@@ -76,23 +76,56 @@ func runCommand(cmd *cobra.Command, source io.ReadCloser) error {
 	var recordRef *corev1.RecordRef
 
 	//nolint:nestif
-	if opts.SignaturePath != "" {
-		signatureSource, err := os.ReadFile(opts.SignaturePath)
-		if err != nil {
-			return fmt.Errorf("failed to read signature file: %w", err)
-		}
-
-		signature := &signv1.Signature{}
-		if err := json.Unmarshal(signatureSource, signature); err != nil {
-			return fmt.Errorf("failed to unmarshal signature: %w", err)
+	if opts.Sign {
```

Comment:
can this logic be moved to a public function under sign command? i dont think we should be doing anything here beyond push operation. signing should be defined under signing logic (ie the command) because then both sign and push commands can reuse that logic